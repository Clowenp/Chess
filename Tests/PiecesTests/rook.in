r 0 0 6 0
r 4 4 6 6
r 7 4 7 8
r 4 0 4 7
