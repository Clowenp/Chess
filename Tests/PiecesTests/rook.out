i am a rook at [0,0]
my possible paths are: 
< [0,1] [0,2] [0,3] [0,4] [0,5] [0,6] [0,7] >
< [1,0] [2,0] [3,0] [4,0] [5,0] [6,0] [7,0] >
< >
< >
the move to [6,0] is possible
i am a rook at [4,4]
my possible paths are: 
< [4,5] [4,6] [4,7] >
< [5,4] [6,4] [7,4] >
< [4,3] [4,2] [4,1] [4,0] >
< [3,4] [2,4] [1,4] [0,4] >
the move to [6,6] is NOT possible
i am a rook at [7,4]
my possible paths are: 
< [7,5] [7,6] [7,7] >
< >
< [7,3] [7,2] [7,1] [7,0] >
< [6,4] [5,4] [4,4] [3,4] [2,4] [1,4] [0,4] >
the move to [7,8] is NOT possible
i am a rook at [4,0]
my possible paths are: 
< [4,1] [4,2] [4,3] [4,4] [4,5] [4,6] [4,7] >
< [5,0] [6,0] [7,0] >
< >
< [3,0] [2,0] [1,0] [0,0] >
the move to [4,7] is possible
